src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index");
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
```

Comment:
```suggestion
        checkCapacity();
        System.arraycopy(elementData, index, elementData, index + 1, size - index);
```
we need to checkCapacity in all `add()` methods, right?